backend/main.go

```diff
@@ -42,20 +45,61 @@ const (
 type Server struct {
 	store        storage.StorageService
 	storeTimeout time.Duration
+	auth         Auth
+}
+
+type Auth interface {
```

Comment:
These auth-related types would probably be best put in a new package, called `auth` or something.

Reply:
Created package auth with auth.go and firebase.go